easybuild/tools/modules.py

```diff
@@ -59,9 +59,10 @@
 VERSION_ENV_VAR_NAME_PREFIX = "EBVERSION"
 DEVEL_ENV_VAR_NAME_PREFIX = "EBDEVEL"
 
-# keep track of original LD_LIBRARY_PATH, because we can change it by loading modules and break modulecmd
+# keep track of original $LD_LIBRARY_PATH/$LD_PRELOAD, because it can change by loading modules and break module command
 # see e.g., https://bugzilla.redhat.com/show_bug.cgi?id=719785
-LD_LIBRARY_PATH = os.getenv('LD_LIBRARY_PATH', '')
+LD_ENV_VAR_KEYS = ['LD_LIBRARY_PATH', 'LD_PRELOAD']
+ORIG_ENVIRON = dict([(key, os.getenv(key, '')) for key in LD_ENV_VAR_KEYS])
```

Comment:
OK, I cleaned this up best I could. We're still copying `os.environ` in a couple of place, but that's at different times in the build process, and the environment may have been changed in the mean time (e.g. after configuring EasyBuild, etc.).
It's a lot easier to simply copy/restore the whole environment at certain points in time, rather than trying to track all changes being made in the environment, and rolling only those changes back when needed...